### PR TITLE
Feature/tfn 452 select passenger type

### PR DESCRIPTION
**Pull request by gilesdrinkall**
# Description

-   To create a page that provides users with the ability to select a passenger type for which they wish to submit fares data.

# Testing instructions

-   The page can be tested locally by running the 'make' command in FDBT-DEV and going to localhost:5555 in the browser.

# Type of change

-   [X] feat - A new feature
-   [ ] fix - A bug fix
-   [ ] docs - Documentation only changes
-   [ ] codestyle - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] refactor - A code change that neither fixes a bug nor adds a feature
-   [ ] perf - A code change that improves performance
-   [ ] test - Adding missing tests or correcting existing tests
-   [ ] build - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
-   [ ] ci - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] chore - Other changes that don't modify src or test files
-   [ ] revert - Reverts a previous commit
-   [ ] content - Changes to content or copy

# Checklist:

-   [X] Able to run pr locally
-   [X] Lighthouse performed
-   [X] Followed acceptance criteria
-   [X] My code follows the style guidelines of this project
-   [X] I have performed a self-review of my own code
-   [ ] I have made corresponding changes to the documentation if applicable
-   [X] I have added tests that prove my fix is effective or that my feature works
Screenshot 2020-05-20 at 15 12 06

### CODE REVIEW

File: src/pages/api/passengerType.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+        if (req.body.passengerType) {
+            const { passengerType } = req.body;
+
+            const cookieValue = JSON.stringify({
+                errorMessage: '',
+                uuid: getUuidFromCookie(req, res),
+                passengerType,
+            });
+            setCookieOnResponseObject(getDomain(req), PASSENGERTYPE_COOKIE, cookieValue, req, res);
```
Comment (tejpowar) [inline src/pages/api/passengerType.ts:38]: add newline

File: src/pages/api/passengerType.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+            if (passengerType === 'Any') {
+                redirectOnFareType(fareType, res);
+                return;
+            }
+            redirectTo(res, '/definePassengerType');
+            return;
+        }
+        const passengerTypeCookieValue = JSON.stringify({
```
Comment (tejpowar) [inline src/pages/api/passengerType.ts:47]: same as above add new line

File: src/constants/index.ts [hunk 3]
```diff
@@ -2,6 +2,8 @@ export const OPERATOR_COOKIE = 'fdbt-operator';
 
 export const FARETYPE_COOKIE = 'fdbt-fareType';
 
+export const PASSENGERTYPE_COOKIE = 'fdbt-passengerType';
```
Comment (tejpowar) [inline src/constants/index.ts:5]: Just a thought would this read better as PASSENGER_TYPE_COOKIE. I know the other types have it but seems quite long

File: src/pages/productDetails.tsx [hunk 4]
```diff
@@ -135,6 +142,7 @@ export const getServerSideProps = (ctx: NextPageContext): { props: ProductDetail
         props: {
             product: !productDetailsCookie ? {} : JSON.parse(productDetailsCookie),
             operator: operatorInfo.operator,
+            passengerType: passengerTypeInfo.passengerType,
```
Comment (tejpowar) [inline src/pages/productDetails.tsx:145]: could we destructure this like `const {passengerType} =  passengerTypeInfo `. Then in the props can just have passengerType?

File: src/pages/singleDirection.tsx [hunk 5]
```diff
@@ -92,7 +95,9 @@ export const getServerSideProps = async (ctx: NextPageContext): Promise<{}> => {
             ) === index,
     );
 
-    return { props: { operator: operatorInfo.operator, lineName, service } };
+    return {
+        props: { operator: operatorInfo.operator, passengerType: passengerTypeInfo.passengerType, lineName, service },
```
Comment (tejpowar) [inline src/pages/singleDirection.tsx:99]: similar to  destructure above

File: tests/pages/api/apiUtils/index.test.ts [hunk 6]
```diff
@@ -48,4 +55,51 @@ describe('apiUtils', () => {
...
+            expect(writeHeadMock).toBeCalledWith(302, { Location: '/serviceList' });
+        });
+        it('should throw error if unexpected fare type is selected', () => {
+            const { res } = getMockRequestAndResponse({}, null, {}, writeHeadMock);
+            const fareType = 'roundAbout';
+
+            expect(() => {
+                redirectOnFareType(fareType, res);
+            }).toThrowError(new Error('Fare Type we expect was not received.'));
+        });
```
Comment (tejpowar) [inline tests/pages/api/apiUtils/index.test.ts:95]: newline between each test

File: src/pages/api/passengerType.ts (hunk 1 above)
Comment (gilesdrinkall) [reply src/pages/api/passengerType.ts:38]: Added a line.

File: src/pages/api/passengerType.ts (hunk 2 above)
Comment (gilesdrinkall) [reply src/pages/api/passengerType.ts:47]: Added a line.

File: src/constants/index.ts (hunk 3 above)
Comment (gilesdrinkall) [reply src/constants/index.ts:5]: Changed cookie to PASSENGER_TYPE_COOKIE.

File: src/pages/productDetails.tsx (hunk 4 above)
Comment (gilesdrinkall) [reply src/pages/productDetails.tsx:145]: Added destructuring for 'passengerType'. Did the same for 'operator' for consistency's sake.

File: src/pages/singleDirection.tsx (hunk 5 above)
Comment (gilesdrinkall) [reply src/pages/singleDirection.tsx:99]: Added destructuring for 'passengerType'. Did the same for 'operator' for consistency's sake.

File: tests/pages/api/apiUtils/index.test.ts (hunk 6 above)
Comment (gilesdrinkall) [reply tests/pages/api/apiUtils/index.test.ts:95]: New lines added.

File: src/pages/service.tsx [hunk 7]
```diff
@@ -63,20 +64,22 @@ export const getServerSideProps = async (ctx: NextPageContext): Promise<{}> => {
...
+    const passengerTypeInfo = JSON.parse(passengerTypeCookie);
 
     const services = await getServicesByNocCode(operatorInfo.nocCode);
 
     if (services.length === 0) {
         throw new Error(`No services found for NOC Code: ${operatorInfo.nocCode}`);
     }
 
-    return { props: { operator: operatorInfo.operator, services } };
+    return { props: { operator: operatorInfo.operator, passengerType: passengerTypeInfo.passengerType, services } };
```
Comment (tejpowar) [inline src/pages/service.tsx:82]: destructure here as well

Comment (gilesdrinkall) [reply src/pages/service.tsx]: Destructured passengerType.

**Pull request merged**